### DISCUSSION TITLE
Nested tag loops with partials behave unexpected

### DISCUSSION THREAD

**Discussion by hanneshoepfnerio** [statamic/cms#6867] [Troubleshooting]
Hello,
I've had a discussion in Discord but I'm assuming that there might be some issues, but I'm also very happy to prove me wrong ;)
So this code on a entry detail page:
```
    <div>Title {{ title }}</div>
    {{ outer = [1] }}
      <div>outer: {{ title }}</div>
        {{ pages = {collection:pages} }}
            <div>pages: {{ title }}</div>
             {{partial:components/my_partial }}
        {{ /pages }}
        <div>/pages: {{ title }}</div>
    {{/outer}}
    <div>/outer: {{ title }}</div>

_my_partial.antlers.html //v1
<div>
 {{# {{title}} #}} 
</div>

_my_partial.antlers.html //v2
<div>
</div>
```
This is outputting the following, if the partial v1 is used with commented out antlers
```
Title Entry Title
outer: Entry Title
pages: Page1
pages: Page2
pages: Page3
pages: Page4
pages: Page5
pages: Page6
/pages: Entry Title
/outer: Page5 // <--- Why? How?
```
What is unexpected is that after the iteration over the pages collection the {{title}} var of the entry is altered – in this case to page5 which is not even the last of the iteration.
When using partial v2, without any antlers the var stays the same, as expected.
Also when i comment out the outer loop.
```
Title Entry Title
outer: Entry Title
pages: Page1
pages: Page2
pages: Page3
pages: Page4
pages: Page5
pages: Page6
/pages: Entry Title
/outer: Entry Title
```

I found a workaround which prevents this behavior using aliasing
```
 <div>Title {{ title }}</div>
    {{ outer = [1] }}
    {{ outer as="test" }} // <--- this is fixing it for some reason
      <div>outer: {{ title }}</div>
        {{ pages = {collection:pages} }}
            <div>pages: {{ title }}</div>
             {{partial:components/my_partial }}
        {{ /pages }}
        <div>/pages: {{ title }}</div>
    {{/outer}}
    <div>/outer: {{ title }}</div>
 ```
But it is a little strange to me.

Please help me understand, or check what is going on there.

Thank you!

**Comment by JohnathonKoster** [comment]
This is definitely interesting! I will look into this. Thanks for the excellent write up!

**Reply by hanneshoepfnerio** [reply]
Sure!

Let me know when there are insights on this topic!

Thanks

**Reply by JohnathonKoster** [reply]
I tracked it down to not restoring the appropriate outer scope once its finished processing a loop. The behavior is pretty nuanced, so won't get into too many details here, but this behavior would only manifest under very specific circumstances.

You can follow the status of the fix with this PR: https://github.com/statamic/cms/pull/6894 🙂

**Reply by hanneshoepfnerio** [reply]
Okay, nice.
Too bad that I ran into it though – I thought I'll loose my mind 🤪
